File: numerical-methods/euler-method.c
/* Euler's Method
*  numerical-methods/euler-method.c
*  By: Jishan Shaikh
*  References: Numerical Methods using C by B. S. Grewal
*/
#include <stdio.h>
#include <math.h>

float df(float x, float y){
  return x+y;
}

int main(){
  float x0, x1, h, y0, y1;
  printf("Enter values of x0, y0, h, and x: ");
  scanf("%f %f %f %f", &x0, &y0, &h, &x);
  x1=x0;
  y1=y0;
  while(1){
    if(x1 > x)
      return;
    y1 += h*df(x1, y1);
    x1 += h;
    printf("When x = %3.1f, y = %4.2f\n", x1, y1);
  }
  return 0;
}
// End program euler-method.c

File: numerical-methods/Python3/euler-method.py
# Euler Method

import os
import sys

def df(a, b):
  return a+b
  
if __name__ == "__name__":
  x0, y0, h, x = input.split()
  x1 = x0
  y1 = y0
  while 1:
    if x1 > x:
      break
    y1 += h * df(x1, y1)
    x1 += h
    # print("When x = %3.1f, y = %4.2f\n", x1, y1);
    
# End of program euler-method.py
